modules/xmpp/JingleSessionPC.js

```diff
@@ -2110,14 +2082,12 @@ export default class JingleSessionPC extends JingleSession {
                     }
 
                     return promise.then(() => {
-                        if (newTrack && newTrack.isVideoTrack()) {
+                        if (newTrack?.isVideoTrack()) {
                             logger.debug(`${this} replaceTrack worker: configuring video stream`);
 
-                            // FIXME set all sender parameters in one go?
-                            // Set the degradation preference on the new video sender.
-                            return this.peerconnection.setSenderVideoDegradationPreference()
-                                .then(() => this.peerconnection.setSenderVideoConstraint())
-                                .then(() => this.peerconnection.setMaxBitRate());
+                            // Configure the video encodings and set the active state, max bitrate and degradation
+                            // preference on them.
+                            return this.setSenderVideoConstraint();
```

Comment:
I think this method should be called differently as it's no longer only about senderVideoConstraints. Then the comment above would not be needed to explain that it's all of the encoding parameters being reconfigured. It's a little bit weird that sometimes this method is called without the actual constraint, I think if it was split into `setSenderVideoConstraint` that takes the actual constraint and then another one which only reconfigures based on the current state it would be a little cleaner.